lib/sumi/inspect.rb

```diff
@@ -26,6 +26,72 @@ def call(object)
 		@buffer
 	end
 
+	def inspect(object)
+		original_object = @original_object
+
+		if original_object == Null
+			@original_object = object
+		else
+			if original_object.equal?(object)
+				push "self"
+				return
+			end
+		end
+
+		@stack.push(object)
+
+		if object.respond_to?(:sumi_inspect)
+			object.sumi_inspect(self)
```

Comment:
This is the only new bit. Mostly, this was just moving the method to make it public.